flexget/plugins/list/ombi_list.py

```diff
@@ -0,0 +1,281 @@
+                          tvdb_id=parent_request.get('tvDbId'),
+                          imdb_id=parent_request.get('imdbId'),
+                          ombi_childrequest_id=child_request.get('id'),
+                          ombi_season_id=season.get('id'),
+                          ombi_status=parent_request.get('status'),
+                          ombi_request_id=parent_request.get('id'))
+        elif self.config.get('type') == 'episodes':
+            log.debug('Episode Number: %s', episode.get('episodeNumber'))
+            entry = Entry(title=self.generate_title(parent_request, season, episode),
+                          url=episode.get('url'),
+                          series_name=self.generate_title(parent_request),
+                          series_season=season.get('seasonNumber'),
+                          series_episode=episode.get('episodeNumber'),
+                          series_id=self.generate_series_id(season, episode),
+                          tvdb_id=parent_request.get('tvDbId'),
+                          imdb_id=parent_request.get('imdbId'),
+                          ombi_request_id=parent_request.get('id'),
+                          ombi_childrequest_id=child_request.get('id'),
+                          ombi_season_id=season.get('id'),
+                          ombi_episode_id=episode.get('id'),
+                          ombi_approved=episode.get('approved'),
+                          ombi_available=episode.get('available'),
+                          ombi_requested=episode.get('requested'))
+        else:
+            raise plugin.PluginError('Error: Unknown list type %s.' % (self.config.get('type')))
+
+        return entry
+
+    @property
+    def items(self):
+        if not self._items:
+
+            json = self.get_request_list()
+
+            self._items = []
+
+            for parent_request in json:
+                if self.config.get('type') == 'movies':
+                    # check that the request is approved unless user has selected to include everything
+                    if self.config.get('only_approved') and not parent_request.get('approved') or parent_request.get('approved'):
+                        # Always include items that are not available and only include available items if user has selected to do so
+                        if (self.config.get('include_available') and parent_request.get('available') or not parent_request.get('available')):
+                            entry = self.generate_movie_entry(parent_request)
+                            log.debug('Entry %s', entry)
+                            self._items.append(entry)
+                elif self.config.get('type') == 'shows':
+                    # Shows do not have approvals or available flags so include them all
+                    entry = self.generate_tv_entry(parent_request)
+                    log.debug('Valid entry %s', entry)
+                    self._items.append(entry)
+                else:
```

Comment:
Consider splitting the contents of the outer-most if-elif-else into separate function